src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -1454,17 +1454,24 @@ module SimulationCode =
                     []
```

Comment:
From @bettinaheim post: Could you please add tests for this?

Reply:
Sure. I've been looking at SimulationCodeTests.fs to get a feel for these kinds of tests, but that file is huge. Is there any tools or tests in particular you could point out to me to help me write a test for this?

Reply:
We should probably split SimulationCodeTests.fs into multiple files. I agree it's way too big.

Reply:
Looking at the code gen test for the piece that you modified makes sense. It would also be good to additionally have some execution tests - there is e.g. a nice case in the bug report, and you should add some more complex cases as well.

Reply:
@bettinaheim I've added a test for the bug fix. As for the execution test, I don't think we need to go that far. This PR is a simple bug fix, after all.

Reply:
@samarsha showed me that writing execution tests is way easier in this case than I thought, so I have added an execution test for this as well.